src/ui/ResultFolding/ResultFolding.ts

```diff
@@ -173,42 +163,37 @@ export class ResultFolding extends Component {
     this.results.appendChild(this.waitAnimation);
     this.updateElementVisibility();
 
-    let ret = this.moreResultsPromise.then((results?: IQueryResult[]) => {
-      this.childResults = results;
-      this.showingMoreResults = true;
-      this.usageAnalytics.logClickEvent<IAnalyticsDocumentViewMeta>(
-        analyticsActionCauseList.foldingShowMore,
-        this.getAnalyticsMetadata(),
-        this.result,
-        this.element
-      );
-      return this.displayThoseResults(results).then(() => {
-        this.updateElementVisibility(results.length);
-        return results;
-      });
-    });
+    const results: IQueryResult[] = await this.moreResultsPromise;
+    this.childResults = results;
+    this.showingMoreResults = true;
+    this.usageAnalytics.logClickEvent<IAnalyticsDocumentViewMeta>(
+      analyticsActionCauseList.foldingShowMore,
+      this.getAnalyticsMetadata(),
+      this.result,
+      this.element
+    );
+    await this.displayThoseResults(results);
```

Comment:
This should probably be wrapped in a try / catch to replicate the same logic that we had before, with the loading animation always properly removed even if the template throws an error ( result templates are kinda like "end users" input since they can be customized, and can throw an error without us having any control over it).

It could also be tested.